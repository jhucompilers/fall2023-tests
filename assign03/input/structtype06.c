void strcpy(char *dst, const char *src);

struct Item {
  char name[32];
  int price;
  int quantity;
};

int main(void) {
  struct Item item;

  // item.name is an array of char elements, and used without
  // a subscript it "decays" to become a pointer to char
  strcpy(item.name, "apples");
  item.price = 120;
  item.quantity = 10;

  return 0;
}
