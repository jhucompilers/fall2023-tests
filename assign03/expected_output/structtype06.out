0|strcpy|function|function (pointer to char, pointer to const char) returning void
0|struct Item|type|struct Item {}
1|name|variable|array of 32 x char
1|price|variable|int
1|quantity|variable|int
0|main|function|function () returning int
2|item|variable|struct Item {array of 32 x char, int, int}
